hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -919,12 +920,32 @@ private boolean isKeyEmpty(OmKeyInfo keyInfo) {
     // underlying table using an iterator. That automatically creates a
     // snapshot of the data, so we don't need these locks at a higher level
     // when we iterate.
+
+    startKey = normalizeListKeyPath(startKey);
+    keyPrefix = normalizeListKeyPath(keyPrefix);
```

Comment:
NIT:

I think it's easier to follow to move the condition to here:

```
     if (enableFileSystemPaths) {
       startKey = normalizeListKeyPath(startKey);
       keyPrefix = normalizeListKeyPath(keyPrefix);
    }
```

Despite the name, the `startKey` and `keyPrefix` are not normalized if `enableFileSystemPaths` is false

Reply:
As anyway this condition needs to be checked inside normalizeListKeyPath, so not performed check again here.

Reply:
Sorry, I don't get it. Why do we need to check it _inside normalizeListKeyPath_.

As I wrote it's a very minor thing, but it seems to be better to move out the check from the method because it improves the readability (IMHO!).

When somebody read the `listKeys` method it suggest the the keys are normalized. but in fact it's normalized only if enableFileSystemPaths. This can be confusing as the method name is `normalizeListKeyPath` and not something like `normalizeListKeyPathIfNormalizationIsEnabled`.

I suggested moving out this condition from this method to improve the readibility, but if you think it's a bad idea, it can be ignored.

Reply:
Thanks to explain it. I understood your point. Updated the code.